src/toolkits/activity_classification/ac_data_iterator.cpp

```diff
@@ -280,6 +280,153 @@ variant_map_type _activity_classifier_prepare_data_impl(const gl_sframe &data,
     return result_dict;
 }
 
+variant_map_type _activity_classifier_prepare_data_aug_impl(const gl_sframe &data,
+                                                        const std::vector<std::string> &features,
+                                                        const std::string &session_id,
+                                                        const int &prediction_window,
+                                                        const int &predictions_in_chunk,
+                                                        const std::string &target,
+                                                        bool verbose) {
+
+    DASSERT_TRUE(features.size() > 0);
+    DASSERT_TRUE(prediction_window > 0);
+    DASSERT_TRUE(predictions_in_chunk > 0);
+    DASSERT_TRUE(data.contains_column(session_id));
+    for (TURI_ATTRIBUTE_UNUSED_NDEBUG auto &feat : features) {
+        DASSERT_TRUE(data.contains_column(feat));
+    }
+
+    bool use_target = (target != "");
+    if (use_target) {
```

Comment:
Similar to the previous point, I would probably replace this entire if-statement with:
`DASSERT_TRUE(use_target && data.contains_column(target));`